ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/LMDB.hs

```diff
@@ -252,42 +268,68 @@ initRangeReadLMDBTable count (LMDBMK _        db)                      = ApplyVa
 rangeReadLMDBTable     count (LMDBMK _        db) (ApplyKeysMK   keys) = ApplyValuesMK    <$> lmdbRangeReadTable count db keys
 
 {-------------------------------------------------------------------------------
- Db settings
+ Db state
 -------------------------------------------------------------------------------}
 
-readDbSettingsMaybeNull ::
+readDbStateMaybeNull ::
      LMDB.Database () DbState
   -> LMDB.Transaction mode (Maybe DbState)
-readDbSettingsMaybeNull db = LMDB.get db ()
+readDbStateMaybeNull db = LMDB.get db ()
 
-withDbSettingsRW ::
+withDbStateRW ::
      LMDB.Database () DbState
   -> (DbState -> LMDB.Transaction LMDB.ReadWrite (a, DbState))
   -> LMDB.Transaction LMDB.ReadWrite a
-withDbSettingsRW db f = withDbSettingsRWMaybeNull db $ maybe (Exn.throw DbErrNoSettings) f
+withDbStateRW db f = withDbStateRWMaybeNull db $ maybe (Exn.throw DbErrNoDbState) f
 
-withDbSettingsRWMaybeNull ::
+withDbStateRWMaybeNull ::
       LMDB.Database () DbState
    -> (Maybe DbState -> LMDB.Transaction LMDB.ReadWrite (a, DbState))
    -> LMDB.Transaction LMDB.ReadWrite a
-withDbSettingsRWMaybeNull db f  =
-  readDbSettingsMaybeNull db >>= f >>= \(r, new_s) -> LMDB.put db () (Just new_s) $> r
+withDbStateRWMaybeNull db f  =
+  readDbStateMaybeNull db >>= f >>= \(r, new_s) -> LMDB.put db () (Just new_s) $> r
 
 data GuardDbDir  = GDDMustExist | GDDMustNotExist
 
-guardDbDir :: MonadIO m => GuardDbDir -> FS.SomeHasFS m -> FS.FsPath -> m FilePath
+-- | Guard for the existence/non-existence of a database directory,
+-- and create it if missing.
+guardDbDir ::
+     (MonadIO m, IOLike m)
+  => GuardDbDir
+  -> FS.SomeHasFS m
+  -> FS.FsPath
+  -> m FilePath
 guardDbDir gdd (FS.SomeHasFS fs) path = do
-  FS.doesFileExist fs path >>= \b -> when b $ Exn.throw $ DbErrStr $ "guardDbDir:must be a directory:" <> show path
-  FS.doesDirectoryExist fs path >>= \case
-    True | GDDMustNotExist <- gdd ->
-           -- TODO Should throw a DbErrDbExists exception
-           -- Callers should delete the directory if they want to restore a snapshot
-           FS.removeDirectoryRecursive fs path
-    False | GDDMustExist <- gdd -> Exn.throw $ DbErrDbDoesntExist path
-    _ -> pure ()
+  fileEx <- FS.doesFileExist fs path
+  when fileEx $
+    Exn.throw $ DbErrStr $ "guardDbDir:must be a directory: " <> show path
```

Comment:
I'm a bit confused when we use `throwIO` and `Exn.throw`. Shouldn't we use the former?

Reply:
You're right, I'm not being consistent with this yet. [Control-Exception:throwIO](https://hackage.haskell.org/package/base-4.14.1.0/docs/Control-Exception.html#v:throwIO) does recommend using `throwIO` instead of `throw` when in `IO`, so I'll do a pass over the code to make the usage of `throwIO` and `throw` consistent

Reply:
We decided to favour using `throwIO` (or `liftIO . throwIO`)